iceoryx2-cal/tests/resizable_shared_memory_tests.rs

```diff
@@ -0,0 +1,954 @@
+    }
+
+    #[test]
+    fn deallocate_removes_unused_segments_on_creator_side<
+        Shm: SharedMemory<DefaultAllocator>,
+        Sut: ResizableSharedMemory<DefaultAllocator, Shm>,
+    >() {
+        let storage_name = generate_name();
+        let config = generate_isolated_config::<Sut>();
+
+        let sut_creator = Sut::MemoryBuilder::new(&storage_name)
+            .config(&config)
+            .max_number_of_chunks_hint(1)
+            .allocation_strategy(AllocationStrategy::PowerOfTwo)
+            .create()
+            .unwrap();
+
+        let ptr_creator_1 = sut_creator.allocate(Layout::new::<u64>()).unwrap();
+        assert_that!(sut_creator.number_of_active_segments(), eq 1);
+
+        let _ptr_creator_2 = sut_creator.allocate(Layout::new::<u64>()).unwrap();
+        assert_that!(sut_creator.number_of_active_segments(), eq 2);
+
+        unsafe { sut_creator.deallocate(ptr_creator_1.offset, Layout::new::<u64>()) };
+        assert_that!(sut_creator.number_of_active_segments(), eq 1);
+    }
+
+    #[test]
+    fn unregister_removes_unused_segments_on_viewer_side<
+        Shm: SharedMemory<DefaultAllocator>,
+        Sut: ResizableSharedMemory<DefaultAllocator, Shm>,
+    >() {
+        let storage_name = generate_name();
+        let config = generate_isolated_config::<Sut>();
+
+        let sut_creator = Sut::MemoryBuilder::new(&storage_name)
+            .config(&config)
+            .max_number_of_chunks_hint(1)
+            .allocation_strategy(AllocationStrategy::PowerOfTwo)
+            .create()
+            .unwrap();
+
+        let ptr_creator_1 = sut_creator.allocate(Layout::new::<u64>()).unwrap();
+        assert_that!(sut_creator.number_of_active_segments(), eq 1);
+
+        let _ptr_creator_2 = sut_creator.allocate(Layout::new::<u64>()).unwrap();
+        assert_that!(sut_creator.number_of_active_segments(), eq 2);
+
+        unsafe { sut_creator.deallocate(ptr_creator_1.offset, Layout::new::<u64>()) };
+        assert_that!(sut_creator.number_of_active_segments(), eq 1);
+    }
+
+    fn allocate_more_than_hinted_with_increasing_chunk_size_works<
+        Shm: SharedMemory<DefaultAllocator>,
+        Sut: ResizableSharedMemory<DefaultAllocator, Shm>,
+    >(
+        strategy: AllocationStrategy,
+    ) {
+        const NUMBER_OF_REALLOCATIONS: usize = 128;
+        let storage_name = generate_name();
+        let config = generate_isolated_config::<Sut>();
+
+        let sut_creator = Sut::MemoryBuilder::new(&storage_name)
+            .config(&config)
+            .max_chunk_layout_hint(Layout::new::<u8>())
+            .max_number_of_chunks_hint(1)
+            .allocation_strategy(strategy)
+            .create()
+            .unwrap();
+
+        let sut_viewer = Sut::ViewBuilder::new(&storage_name)
+            .config(&config)
+            .open()
+            .unwrap();
+
+        let mut ptrs = vec![];
+        for i in 0..NUMBER_OF_REALLOCATIONS {
+            let size = 2 + i;
```

Comment:
I you really want the `NUMBER_OF_REALLOCATIONS` be the real number of reallocations, even for power of two, then the size should probably be `2 * i + 1` and run from `1..=NUMBER_OF_REALLOCATIONS`

Reply:
But with a power of two strategy this will explode. And just for fun, let's assume we have a power of hundred strategy then this will be hard to test for nearly any number.

I renamed the variable into `NUMBER_OF_ITERATIONS`. The underlying strategy will define how many reallocations will happen but this is not what we want to test here. We just want to ensure that the chunk increases.

Reply:
Right. My proposal was also flawed since it did not significantly increase the number of re-allocations.

Renaming the constant helped.